adapters/adapters-test-utils/src/main/java/uk/gov/justice/services/adapters/test/utils/builder/HttpActionBuilder.java

```diff
@@ -0,0 +1,174 @@
+        HttpActionBuilder httpActionBuilder = new HttpActionBuilder()
+                .withHttpActionType(actionType)
+                .with(defaultMapping());
+        for (String mimeType : mimeTypes) {
+            httpActionBuilder = httpActionBuilder.withMediaType(mimeType);
+        }
+        return httpActionBuilder;
+    }
+
+    public static HttpActionBuilder httpActionWithNoMapping(final ActionType actionType, final String... mimeTypes) {
+        HttpActionBuilder httpActionBuilder = new HttpActionBuilder()
+                .withHttpActionType(actionType);
+        for (String mimeType : mimeTypes) {
+            httpActionBuilder = httpActionBuilder.withMediaType(mimeType);
+        }
+        return httpActionBuilder;
+    }
+
+    public HttpActionBuilder withHttpActionType(final ActionType actionType) {
+        this.actionType = actionType;
+        return this;
+    }
+
+    public HttpActionBuilder withHttpActionOfDefaultRequestType() {
+        return withMediaType("application/vnd.ctx.command.defcmd+json")
+                .with(mapping()
+                        .withName("action1")
+                        .withRequestType("application/vnd.ctx.command.defcmd+json"));
+    }
+
+    public HttpActionBuilder withDefaultResponseType() {
+        return withResponseTypes("application/vnd.ctx.query.defquery+json")
+                .with(mapping()
+                        .withName("action1")
+                        .withResponseType("application/vnd.ctx.query.defquery+json"));
+    }
+
+    public HttpActionBuilder withResponseTypes(final String... responseTypes) {
+        return withHttpActionResponse(new Response(), responseTypes);
+    }
+
+    public HttpActionBuilder withHttpActionResponse(final Response response, final String... responseTypes) {
+        Map<String, MimeType> respBody = new HashMap<>();
+        for (String responseType : responseTypes) {
+            respBody.put(responseType, new MimeType(responseType));
+        }
+        response.setBody(respBody);
+        responses.add(response);
+        return this;
+    }
+
+    public HttpActionBuilder withQueryParameters(final QueryParameter... queryParameters) {
+        stream(queryParameters).forEach(queryParameter -> this.queryParameters.put(queryParameter.getDisplayName(), queryParameter));
+        return this;
+    }
+
+    public HttpActionBuilder withQueryParameters(final String... paramNames) {
+        stream(paramNames).forEach(paramName -> {
+            QueryParameter queryParameter = new QueryParameter();
+            queryParameter.setDisplayName(paramName);
+            queryParameter.setType(STRING);
+            queryParameter.setRequired(true);
+            this.queryParameters.put(paramName, queryParameter);
+        });
+
+        return this;
+    }
+
+    public HttpActionBuilder withOptionalQueryParameters(final String... paramNames) {
```

Comment:
Can you collapse these into a single prviate method with just a required flag as that seems to be the only difference.